### PR TITLE
find and report on duplicate modules

### PR DESCRIPTION
**Pull request by laraPPr**

### CODE REVIEW

**Comment by laraPPr** [comment]
Possible fix for https://github.com/EESSI/test-suite/issues/132

**Comment by laraPPr** [comment]
Gives a warning and does not create any tests if it find duplicate modules for a test.
So if one duplicate module of LAMMPS is found it will not create any LAMMPS tests but it will create the other tests (at least if they don't have any duplicate tests).
The error includes a list of the duplicate modules that triggered the error.

File: eessi/testsuite/utils.py [hunk 1]
```diff
@@ -109,8 +111,18 @@ def find_modules(regex: str, name_only=True) -> Iterator[str]:
...
+            if orig_mod in seen:
+                dupes.append(orig_mod)
+            else:
+                seen.add(orig_mod)
             yield orig_mod
 
+    if dupes != []:
+        err_msg = "EESSI test-suite cannot handle duplicate modules."
+        err_msg += "Please make sure that only one is available on your system."
+        err_msg += f"The following modules have a duplicate on your system: {dupes}"
```
Comment (smoors) [inline eessi/testsuite/utils.py:123]:
```suggestion
    if dupes:
        err_msg = "EESSI test-suite cannot handle duplicate modules. "
        err_msg += "Please make sure that only one is available on your system. "
        err_msg += f"The following modules have a duplicate on your system: {dupes}"
```

**Review by smoors — COMMENTED** [review]
looks good, just a few style suggestions

**Review by smoors — APPROVED** [review]
tested and works well, thanks a lot @laraPPr !

**Pull request merged**